getting latest changes

mallikarjunece opened the pull request:
moving to top

buildhive commented:
[Mallikarjuna » jenkins #2](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/2/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #3](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/3/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #4](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/4/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #5](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/5/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #6](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/6/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #7](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/7/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #8](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/8/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #9](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/9/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #10](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/10/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #11](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/11/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #12](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/12/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #13](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/13/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #14](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/14/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #15](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/15/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #16](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/16/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #17](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/17/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #18](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/18/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #19](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/19/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #20](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/20/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #21](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/21/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #22](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/22/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #23](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/23/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #24](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/24/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #25](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/25/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #26](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/26/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #27](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/27/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #28](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/28/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #29](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/29/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #30](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/30/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #31](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/31/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #32](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/32/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #33](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/33/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #34](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/34/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #35](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/35/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #36](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/36/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #37](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/37/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #38](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/38/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #39](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/39/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #40](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/40/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #41](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/41/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #42](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/42/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #43](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/43/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #44](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/44/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #45](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/45/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #46](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/46/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #47](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/47/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #48](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/48/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #49](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/49/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #50](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/50/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #51](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/51/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #52](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/52/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #53](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/53/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #54](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/54/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #55](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/55/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #56](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/56/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #57](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/57/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #58](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/58/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #59](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/59/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #60](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/60/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #61](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/61/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #62](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/62/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #63](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/63/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Mallikarjuna » jenkins #64](https://buildhive.cloudbees.com/job/mallikarjunece/job/jenkins/64/) ABORTED

[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

oleg-nenashev committed:
[JENKINS-9283] - Document timezones support in help pages

oleg-nenashev committed:
[JENKINS-9283] - Don't accept invalid timezones, add form validation

alshopov committed:
Bulgarian translation of properies files

alshopov committed:
Bulgarian translation of HTML files

istrangiu committed:
JENKINS-42854: Added description field to the 'Computer' api

oleg-nenashev committed:
Address comment from @daniel-beck

oleg-nenashev committed:
Merge pull request #2896 from oleg-nenashev/bug/JENKINS-44103-UI

[JENKINS-44103] - Cleanup usages of the "JNLP" term in TcpSlaveAgentListener

oleg-nenashev committed:
Annotate and document methods of hudson.Launcher (#2920)

* Annotate and document methods of hudson.Launcher

* Fix newly introduced FindBugs issues in hudson.Launcher

* Remove brackets from FilePath#createLauncher() to avoid the merge conflict

daniel-beck committed:
Merge pull request #2921 from jenkinsci/pr-template-update

Simplify the pull-request template

daniel-beck committed:
Merge pull request #2895 from abayer/jenkins-34464

[JENKINS-34464] Switch ReverseBuildTrigger.threshold to DataBoundSetter

daniel-beck committed:
Merge pull request #2924 from jglick/annotation-indexer-1.12

annotation-indexer 1.12

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.67

oleg-nenashev committed:
[JENKINS-38527] - Prevent NullPointerException in Slave#createLauncher() and add cause diagnostics (#2923)

* [JENKINS-38527] - Prevent NullPointerException in Slave#createLauncher() and add cause diagnostics

The original issue comes from the isUnix() unboxing, but we can also get into an issue later if we pass a null Channel instance to the logic.
This change adds some diagnostics which discovers potential root causes of such potential NPEs due to the race conditions with Computer reconnection

* [JENKINS-38527] - Also handle cases when Channel#isClosingOrClosed() as @stephenc suggested

daniel-beck committed:
Add list of supported time zone IDs, rephrase some of the doc

jglick committed:
[INFRA-1032] ${user.name} might contain metacharacters.

jglick committed:
The `reload-configuration` CLI command ought to wait until the reload is finished.

oleg-nenashev committed:
Merge pull request #2929 from jglick/user.name.escaped

[INFRA-1032] ${user.name} might contain metacharacters

daniel-beck committed:
s/timezone/time zone

olivergondza committed:
Merge pull request #2897 from olivergondza/JENKINS-44361

[FIXED JENKINS-44361] Follow HTTP redirects while initiating CLI connection

oleg-nenashev committed:
[JENKINS-39370] - Update Remoting in Jenkins core to 3.10 (#2886)

* Update Remoting in Jenkins core to 3.8

* JENKINS-39370 - Introduce support of Work Directories in remoting (opt-in).
* PR 129 - Allow configuring java.util.logging settings via a property file (-loggingConfig or JUL system property). See the Logging page for more details.
* JENKINS-37567 - Change of the code signing certificate

More info: https://github.com/jenkinsci/remoting/blob/master/CHANGELOG.md#38

* [JENKINS-39370] - Add direct tests for JNLP Launcher start with -workDir

* Pick Remoting 3.9

* Improve error message of LauncherTest#remoteKill()

* Update Remoting to 3.10

oleg-nenashev committed:
[JENKINS-43496] - Add handling of the null Node#createComputer() result. (#2922)

* [JENKINS-43496] - Add handling of the null Node#createComputer() result.

it is a follow-up to https://github.com/jenkinsci/jenkins/pull/2836#discussion_r110604865

De-facto many Cloud plugins return `null` in `Node#createLauncher()`, but it has never been documented.
In order to prevent possible API misusages in the future, I have added annotations and fixed handling of the extension point in `AbstractCIBase#updateComputer()` which may fail in the case of `null` or `RuntimeException` in the Node implementation.

* [JENKINS-43496] - Use ProtectedExternally to protect Node#createComputer()

* [JENKINS-43496] - Remove the erroneous Nonnull annotation after the feedback from @jglick

* [JENKINS-43496] - Fix typos noticed by @daniel-beck

oleg-nenashev committed:
Merge pull request #2927 from daniel-beck/JENKINS-9283-docs

Document timezone specification and provide full list

oleg-nenashev committed:
Merge pull request #2931 from jglick/ReloadConfigurationCommand

[JENKINS-45256] The reload-configuration CLI command ought to wait until the reload is finished

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.68

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

jglick committed:
[JENKINS-39738, JENKINS-43668] - Pick up SSHD module including sshd-core 1.6.0 (#2853)

* Pick up SSHD module including sshd-core 1.4.0.

* [FIXED JENKINS-39738] Picking up https://github.com/jenkinsci/sshd-module/commit/bb6963453086edef4e600abb2740182cd2f4ade0

* Picking up Apache SSHD 1.6.0 & Jenkins sshd module 2.0.

oleg-nenashev committed:
[JENKINS-34914] - Prevent NPE in Jenkins#getRootURL() when the nstance is not fully loaded

nfalco79 committed:
[JENKINS-14807] Fix path separator when EnvVars overrides variable like
PATH+XYZ

The getEnvironment(Node, TaskListener) now returns an environment setup
correctly with the platform value of the computer node where the job is
executed.

olamy committed:
last winstone for http2 connector

Signed-off-by: olivier lamy <olamy@apache.org>

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.69

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

olamy committed:
use new executable-war to fix classloading issue with services loader

Signed-off-by: olivier lamy <olamy@apache.org>

daniel-beck committed:
Show actual Jenkins version on upgrade wizard banner

daniel-beck committed:
[JENKINS-45459] Fix 'shortened' Jenkins version in upgrade wizard

daniel-beck committed:
Merge pull request #2938 from daniel-beck/flexible-upgrade-wizard

Show actual Jenkins version in upgrade wizard

jglick committed:
Deleting obsolete SECURITY-144-compat exclusion.

jcmoore3 committed:
[JENKINS-45516] Fix null pointer exception when checking for previous completed build

daniel-beck committed:
Fix @since tags

daniel-beck committed:
Fix more since tags

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.70

olamy committed:
executable-war 1.36

Signed-off-by: olivier lamy <olamy@apache.org>

oleg-nenashev committed:
Merge pull request #2935 from oleg-nenashev/bug/JENKINS-34914-getRootUrl()-NPE

[JENKINS-34914] - Prevent NPE in Jenkins#getRootURL() when the instance is not fully loaded

daniel-beck committed:
Merge pull request #2943 from daniel-beck/fix-since

Fix @since tags

olamy committed:
no need anymore to ignore jetty spdy classes as it doesn't exist anymore

Signed-off-by: olivier lamy <olamy@apache.org>

daniel-beck committed:
[FIX JENKINS-45679] JNLP needs to request Java 8

olamy committed:
use release 4.1 of winstone

Signed-off-by: olivier lamy <olamy@apache.org>

olamy committed:
Merge pull request #2937 from olamy/feature/jetty_http2_connector

[JENKINS-45438] Add Jetty HTTP/2 connector

oleg-nenashev committed:
Merge pull request #2936 from nfalco79/feature/JENKINS-14807

[JENKINS-14807] Fix path separator when EnvVars overrides variable like PATH+XYZ on slave node cross platform

oleg-nenashev committed:
Merge pull request #2944 from daniel-beck/JENKINS-45679

[FIX JENKINS-45679] JNLP needs to request Java 8

oleg-nenashev committed:
Merge pull request #2941 from jcmoore3/JENKINS-45516

[JENKINS-45516] Fix null pointer exception when checking for previous completed build

oleg-nenashev committed:
Merge pull request #2940 from jglick/cleanup-JENKINS-25625

[JENKINS-25625] Deleting obsolete SECURITY-144-compat exclusion

jglick committed:
[JENKINS-45737] - User mapping should be stored in a per-Jenkins field and getAll should be called just once per session unless we reload (#2928)

[JENKINS-45737] - User mapping should be stored in a per-Jenkins field and getAll should be called just once per session unless we reload

arothian committed:
[JENKINS-45519] Fix keepUndefinedParameters option for suppressing warnings (#2939)

* Update ParametersAction.java

Pull request 2687 added a 'false' setting for this flag to prevent warning messages from being logged, but the logic doesn't match the message or the documentation. This updates the check, so that the warning message is correctly suppressed if `hudson.model.ParametersAction.keepUndefinedParameters` is set to false.

* Updated check to use optBoolean

damianszczepanik committed:
Added Polish translations for setup wizard

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.71

Commit:
Fixed typo in German translation.

daniel-beck committed:
Merge pull request #2946 from Jochen-A-Fuerbacher/fixTypo

Fixed typo in German translation.

jglick committed:
Avoid calling Task.getFullDisplayName unless and until we need to.

daniel-beck committed:
[JENKINS-38185] Always follow redirects for DownloadService

oleg-nenashev committed:
[JENKINS-44112] - Enable WorkDir in JNLPLauncher (#2945)

* [JENKINS-44112] - Enable Work Directories by defaul in new agents with JNLP launcher

* [JENKINS-44112] - Add test for the legacy data migration

* [JENKINS-44112] - Add test for API calls

* [JENKINS-44112] - Apply new APIs within core and tests

* [JENKINS-44112] - Cleanup tests

* [JENKINS-44112] - Cleanup API of JNLPLauncher

* [JENKINS-44112] - Address the leftover code review comments

oleg-nenashev committed:
Merge pull request #2951 from daniel-beck/JENKINS-38185

[JENKINS-38185] Always follow redirects for DownloadService

oleg-nenashev committed:
Merge pull request #2947 from jglick/Queue-opt

[JENKINS-45553] Avoid calling Task.getFullDisplayName unless and until we need to

jglick committed:
[JENKINS-45553] - Cache the Mac so we do not need to constantly recreate it (#2948)

[JENKINS-45553] - Cache the Mac so we do not need to constantly recreate it

abayer committed:
[FIXED JENKINS-20272] Don't monitor response on offline agents (#2911)

* [FIXED JENKINS-20272] Don't monitor response on offline agents

* Updating to only not check if channel is null.

* Fix broken test.

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.72

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

jglick committed:
[JENKINS-45892] Prevent a Run from being serialized except at top level.

jglick committed:
Extending fix to AbstractItem.

jglick committed:
Extending fix to User.

jglick committed:
Using RestartableJenkinsRule.then.

ndeloof committed:
add missing @Symbol for agent protocols

jglick committed:
jenkins-test-harness 2.24

oleg-nenashev committed:
[FIXED JENKINS-45895] - JNLPLauncher config.jelly should not display Work Dir settings when included from other class

jglick committed:
Merge branch 'master' into backref-JENKINS-45892

jglick committed:
Strengthening tests a bit.

jglick committed:
Make readResolve methods tolerate Jenkins.instance == null.

jglick committed:
Simplifying writeReplace methods by factoring common logic into XmlFile.replaceIfNotAtTopLevel.

jglick committed:
Windows builds have been timing out; giving them an extra hour.

jglick committed:
Amend JENKINS-24110 fix to more politely specify what was not being run.

jglick committed:
[JENKINS-43199] Reliably close build log file (#2954)

* [JENKINS-43199] Reliably close build log file.

* Try harder to ensure that the logger is closed at the end of the build.

oleg-nenashev committed:
Merge pull request #2952 from damianszczepanik/wizard

Added Polish translations for setup wizard

jglick committed:
Use official version of jbcrypt 0.4.

svanoort committed:
[JENKINS-45915] - Switch to the Java8 Base64 rather than Apache commons (#2958)

jglick committed:
Merge branch 'master' into lambdas

oleg-nenashev committed:
Update Slave Installer Module to 1.6

* [PR #1](https://github.com/jenkinsci/slave-installer-module/pull/1/) - Cleanup issues reported by FisndBugs
* [JENKINS-42841](https://issues.jenkins-ci.org/browse/JENKINS-42841) - Replace the "slave" term by "agent" where possible

Changelog: https://github.com/jenkinsci/slave-installer-module/blob/master/CHANGELOG.md#16

abayer committed:
[FIXED JENKINS-45909] ReverseBuildTrigger.upstreamProjects should be null safe

oleg-nenashev committed:
Merge pull request #2960 from ndeloof/master

add missing @Symbol for agent protocols

oleg-nenashev committed:
Merge pull request #2962 from oleg-nenashev/bug/JENKINS-45895

[FIXED JENKINS-45895] - JNLPLauncher config.jelly should not display Work Dir settings when included from other class

oleg-nenashev committed:
Merge pull request #2964 from jglick/jbcrypt

Use official version of jbcrypt 0.4 in Hudson Private Security Realm

alvarolobato committed:
[JENKINS-45244] Avoid locking every time getActions is called (#2933)

* [JENKINS-45244] Avoid locking every time getActions is called

* [JENKINS-45244] - Remove initializer

* Add comment clarifying that the actions must be volatile

oleg-nenashev committed:
Merge pull request #2963 from jglick/Executor-warning

Amend JENKINS-24110 fix to more politely specify what was not being run

nfalco79 committed:
Fix italian description for ToolInstallation

Fix the italian translation for the description label for ToolInstallation

daniel-beck committed:
Minor fixes to the title of a view:

- Use display name rather than name
- Only use the special 'Dashboard' label outside any folder

bkmeneguello committed:
Update Messages_pt_BR.properties

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.73

damianszczepanik committed:
Added Polish translations

oleg-nenashev committed:
Merge pull request #2965 from oleg-nenashev/slave-installer/1.6

Update Slave Installer Module to 1.6

jglick committed:
[JENKINS-45892] Merged #2957: prevent model objects from being serialized except at top level.

olivergondza committed:
[JENKINS-21905] Reproduce in unittest

olivergondza committed:
[JENKINS-21905] Unwrap the unnecessary ExecutionExceptions

olivergondza committed:
[JENKINS-21905] Do not rewrap possibly more specific exception into a less specific one

olivergondza committed:
[JENKINS-21905] Take advantage of suppressed exceptions

olivergondza committed:
[JENKINS-21905] Do not wrap problem with reading into an exception indicating write problem

olivergondza committed:
[JENKINS-21905] Diagnose the known cause without printing stacktrace

olivergondza committed:
[JENKINS-21905] Log the diagnosed cause just in case

olivergondza committed:
[JENKINS-21905] Skip unittest on windows

jglick committed:
[JENKINS-41631] Enforce upper bound deps on Jenkins core (#2956)

* [JENKINS-41631] Enforce upper bound deps on Jenkins core.

* stapler 1.252

abayer committed:
[FIXED JENKINS-46082] API will include culprits again.

daniel-beck committed:
Merge pull request #2971 from bkmeneguello/pt-br-message-fix

Update Messages_pt_BR.properties

oleg-nenashev committed:
Merge pull request #2968 from nfalco79/patch-1

Fix italian description for ToolInstallation

oleg-nenashev committed:
Merge pull request #2974 from damianszczepanik/translations

Added Polish translations

godfath3r committed:
[JENKINS-42376] - Add executor name on Unexpected exception. (#2970)

* Add executor name on Unexpected exception.

* Add a colon after job name

oleg-nenashev committed:
Merge pull request #2978 from abayer/jenkins-46082

[FIXED JENKINS-46082] API will include culprits again.

josiahhaswell committed:
[FIX JENKINS-43848] -  Lack of cache-invalidation headers results in stale item list (#2973)

* Saving progress for review

* Adding licenses

* Adding integration test for headers

* Removing proposal for refactoring to method objects

* Removing whitespace changeswq

* Fixing test

oleg-nenashev committed:
Merge pull request #2969 from daniel-beck/view-title

Minor fixes to the title of a view

oleg-nenashev committed:
Merge pull request #2966 from abayer/jenkins-45909

[FIXED JENKINS-45909] ReverseBuildTrigger.upstreamProjects should be null safe

pbe-axelor committed:
Fixed French translation

Use unicode characters in build_with_parameters french translation.

olamy committed:
Merge pull request #2980 from pbe-axelor/contrib

Fixed French translation.
Merci! :-)

joyyc committed:
Merge remote-tracking branch 'upstream/master'

joyyc committed:
Get rid of excess semicolons

jglick committed:
javadoc:javadoc could fail under some circumstances.

kohsuke committed:
Merge pull request #2982 from jglick/extra-enforcer-rules

javadoc:javadoc could fail under some circumstances

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.74

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

bale836 committed:
[JENKINS-46288] - Fix ProxyConfiguration validation for NTLM authentication (#2984)

/**
     * Constructor.
     * @param userName The user name.  This should not include the domain to authenticate with.
     * For example: "user" is correct whereas "DOMAIN\\user" is not.
     * @param password The password.
     * @param host The host the authentication request is originating from.  Essentially, the
     * computer name for this machine.
     * @param domain The domain to authenticate within.
     */

oleg-nenashev committed:
Merge pull request #2981 from joyyc/master

Just get rid of excess semicolons in h.m.Run

oleg-nenashev committed:
[JENKINS-45841] - Disable JNLP/JNLP2/CLI protocols on new installations (#2950)

* [JENKINS-45841] - Disable JNLP/JNLP2/CLI protocols in the Setup Wizard

* [JENKINS-45841] - Implement deprecation data model for Agent protocols.

WiP

* [JENKINS-45841] - DeprecationCause is YAGNI, extend UI and document deprecated protocols

* [JENKINS-45841] - Fix the layouts

* [JENKINS-45841] - Add administrative monitor for deprecated protocols

* [JENKINS-45841] - Added Initializer check, polished the warning message

* [JENKINS-45841] - Add tests

* [JENKINS-45841] - Replace JNLP protocol links by jenkins.io redirects

* [JENKINS-45841] - Address comments from @jglick

* [JENKINS-45841] - Initializer checks status when Jenkins instance is not ready

oleg-nenashev committed:
[JENKINS-46282] - Update WinSW from 2.1.0 to 2.1.2 (#2987)

* [JENKINS-46282] - Update WinSW from 2.1.0 to 2.1.2

Fixes [JENKINS-46282](https://issues.jenkins-ci.org/browse/JENKINS-46282), which impacts the default installation.
Also updates Parent POM in the module

Full list of fixes:

- JENKINS-46282 - Runaway Process Killer extension was not using the stopTimeoutMs parameter
- [WinSW Issue #206](https://github.com/kohsuke/winsw/issues/206) - Prevent printing of log entries in the `status` command
- [WinSW Issue #218](https://github.com/kohsuke/winsw/issues/218) - Prevent hanging of the stop executable when its logs are not being drained do the parent process

Full Diff: https://github.com/kohsuke/winsw/compare/winsw-v2.1.0...winsw-v2.1.2

* [JENKINS-46282] - Pick the released version of Windows Agent Installer

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.75

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

Jimilian committed:
[JENKINS-29537] EnvironmentContributingAction compatible with Workflow (#2975)

* [JENKINS-29537] EnvironmentContributingAction compatible with Workflow

+ Adds new method with default implementation in EnvironmentContributingAction to support Runs
+ Marks AbstractBuild as deprecated
+ Adds default implementation for deprecated method for backward
compatiblity.

* Tiny improvements in javadoc

oleg-nenashev committed:
Merge pull request #2983 from alshopov/master

Bulgarian translation

oleg-nenashev committed:
Update to Jenkins Parent POM 1.38 (#2985)

* Update to Jenkins Parent POM 1.39

* Pick the released version of Jenkins POM

jglick committed:
[JENKINS-29537] Merged #2993: amended EnvironmentContributingAction signature.

oleg-nenashev committed:
Upgrade Remoting to 3.11 (#2988)

* Use ClassFilter.appendDefaultFilter.

* FindBugs

* Update Jenkins Remoting to 3.11, fix reported FindBugs issues

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.76

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Default the built-in Jenkins Update Center URL to https://updates.jenkins.io

Now that we're using JDK8, we can rely on our Let's Encrypt-based certificates
on *.jenkins.io

Live from Jenkins World!

Signed-off-by: M. Allan <max.allan@surevine.com>
Signed-off-by: R. Tyler Croy <tyler@monkeypox.org>

batmat committed:
[JENKINS-46603] Verify https://github.com/jenkinsci/pom/pull/16

batmat committed:
[JENKINS-46603] Remove overrides to inherit upgraded versions

hplatou committed:
[JENKINS-13153] - Use directory from env:BASE when writing jenkins.copies (#2992)

[JENKINS-13153] - Use directory from env:BASE when writing jenkins.copies

oleg-nenashev committed:
Merge pull request #2996 from rtyler/update-center-https

Default the built-in Jenkins Update Center URL to https://updates.jenkins.io

ndeloof committed:
rely on java8 default methods to avoid code duplication

Signed-off-by: Nicolas De Loof <nicolas.deloof@gmail.com>

jglick committed:
[JENKINS-45892] Enhanced diagnostics (#2997)

* [JENKINS-45892] Enhanced diagnostics.

* Refined fix which should avoid a needless warning when called from MultiBranchProject.onLoad.

Commit:
Rss Bar and Legend Link (Job List Footer) Added Classes and IDs to enable easy styling for external themes (#2989)

* ui classes to enable easy styling

* fix align right html to css

* move css to style.css

* Revert "move css to style.css"

This reverts commit f26162a0f350886040935811d6194d585f8a1bf9.

* move css to style.css (without unrelated spaces changed)

* remove ids and use classes

josiahhaswell committed:
[FIXED JENKINS-31068] Monitor does not detect when Tomcat URL encoding parameter rejects forward slashes in URL  (#2977)

* Fixing JENKINS-31068

* backing out changes--they don't fully work

* Saving progress so that I can revert to an earlier version for tests

* So, pretty exhaustive testing yields that these modifications have the same behavior as the previous versions

* [FIX JENKINS-31068] Adding wiki reference to error message.  Adding trailing slash to URL

* [FIX JENKINS-31068] It looks like different versions of Tomcat and Apache HTTP handle this case differently. Really, the best we can do is check to see if the test method was not hit and passed correctly--if we hit it, we get more information on the configuration error.  If we don't, we just refer them to a general wiki page

ksenia-nenasheva committed:
[FIXED JENKINS-41637] Users with discover permission get broken UI when renaming job to inaccessible one (#2910)

* [JENKINS-41637] - Add new check that new job name isn't used

* Fixed bugs

* Address comment

* [JENKINS-41637] - Address comments

* Add logging

* Address comment

daniel-beck committed:
Merge pull request #2999 from ndeloof/java8default

rely on java8 default methods to avoid code duplication

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.77

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

liketic committed:
[Fix JENKINS-46161] Make ReverseBuildTrigger#getUpstreamProjects null-safe

RandomSort committed:
Update manage_da.properties, fix typo

There was a simple typo. Andministrer is not a word in danish, while administrer is.

liketic committed:
[Fix JENKINS-46161] Add javadoc and annotation CheckForNull

jglick committed:
Merge branch 'master' into lambdas

jglick committed:
[JENKINS-46659] Suppressing failing CLIActionTest.interleavedStdio.

jglick committed:
Merge pull request #2959 from jglick/lambdas

Using RestartableJenkinsRule.then

batmat committed:
Bump to 1.39 release

ndeloof committed:
prefer ‘agent.jar’ over ‘slave.jar’

ndeloof committed:
introduce AGENTJAR_URL
+ minor fixes

ndeloof committed:
fix for Russian translation

olivergondza committed:
[JENKINS-21905] fix typo

daniel-beck committed:
Merge pull request #3001 from RandomSort/master

Update manage_da.properties, fix typo

daniel-beck committed:
Make Computer a DescriptorByNameOwner

This will allow Computers to be used as AncestorInPath e.g. for
form validation.

oleg-nenashev committed:
Merge pull request #3004 from ndeloof/agent

[JENKINS-35451] - prefer ‘agent.jar’ over ‘slave.jar’

batmat committed:
Update to animal-sniffer 1.16 (#2995)

* Update to latest pom for animal-sniffer 1.16

* Bump to 1.39 release

Commit:
[Fix JENKINS-44663] Make Basic authorization header to be case in-sensitive (#3002)

[Fix JENKINS-44663] Make Basic authorization header to be case in-sensitive

daniel-beck committed:
Use default implementation for this interface

jglick committed:
[FIXED JENKINS-46659] Avoid Jetty timeouts by sending a ping-like packet every 10s while a CLI command is running.

rysteboe committed:
[JENKINS-38711] Add UncaughtExceptionHandler to remoting related threads (#2986)

* [JENKINS-38711] Add UncaughtExceptionHandler to remoting related threads

* [JENKINS-38711] Restart listener thread if it fails

* [JENKINS-38711] Reschedule the thread on a delay

* [JENKINS-38711] Allow the listener to actually be restarted

* [JENKINS-38711] Minor updates based on feedback

oleg-nenashev committed:
Revert "[JENKINS-38711] Add UncaughtExceptionHandler to remoting related threads (#2986)"

This reverts commit 50bec3f68426b97344e1b87e4ef9a930f29ec778.

oleg-nenashev committed:
Merge pull request #3013 from jenkinsci/revert-2986-JENKINS-38711

Revert "[JENKINS-38711] Add UncaughtExceptionHandler to remoting related threads"

daniel-beck committed:
Merge pull request #3009 from daniel-beck/Computer-DescriptorByNameOwner

Make Computer a DescriptorByNameOwner

batmat committed:
Use assembly:single instead of deprecated assembly:attached

As attached got *removed* from m-assembly-p:3.x

daniel-beck committed:
Require at least Java 8u101 for the updates.jenkins.io certs

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.78

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

jglick committed:
Suppress compiler warnings about non-API classes used by gadgets.

olivergondza committed:
Trivial doc tweaks in dropdownDescriptorSelector.jelly

jglick committed:
jenkins-test-harness 2.26

olivergondza committed:
Merge pull request #3023 from olivergondza/triavial-doc-tweaks

Trivial doc tweaks in dropdownDescriptorSelector.jelly

oleg-nenashev committed:
[JENKINS-45755, JENKINS-46140] - Update Remoting to 3.12

https://github.com/jenkinsci/remoting/blob/master/CHANGELOG.md#312

jglick committed:
[FIXED JENKINS-46754] Remove org.mindrot:jbcrypt:0.4 since we already bundle org.connectbot.jbcrypt:jbcrypt:1.0.0.

jglick committed:
Consolidate HudsonPrivateSecurityRealmTest & HudsonPrivateSecurityRealm2Test.

jglick committed:
Merge pull request #3026 from jglick/jbcrypt-JENKINS-46754

[JENKINS-46754] Remove org.mindrot:jbcrypt:0.4 since we already bundle org.connectbot.jbcrypt:jbcrypt:1.0.0

oleg-nenashev committed:
Merge pull request #3025 from oleg-nenashev/remoting/3.12

[JENKINS-45755, JENKINS-46140] - Update Remoting to 3.12

oleg-nenashev committed:
Merge pull request #3000 from liketic/JENKINS-46161

[Fix JENKINS-46161] Make ReverseBuildTrigger#getUpstreamProjects null…

olivergondza committed:
[JENKINS-46680] Disconnect computer on ping timeout (#3005)

* [JENKINS-46680] Reproduce in unittest

* [FIX JENKINS-46680] Reset SlaveComputer channel before closing it on ping timeout

* [JENKINS-46680] Attach channel termination offline cause on ping timeouts

oleg-nenashev committed:
Merge pull request #3015 from daniel-beck/letsencrypt

Require at least Java 8u101 for the updates.jenkins.io certs

oleg-nenashev committed:
Merge pull request #2807 from istrangiu/AddApiNodeDescription

JENKINS-42854: Added description field to the 'Computer' api

daniel-beck committed:
[JENKINS-45472] Copy the list in case it's reused elsewhere

batmat committed:
Downgrade restriction from DoNotUse to NoExternalUse

There's a failure with accmod using DoNotUse. So taking the
easy path here to move forward. This will already have the intended
effect: clarify that TimeUnit2 is not to be used anymore.

batmat committed:
Merge pull request #2998 from batmat/JENKINS-46603

[JENKINS-46603] Bump to latest parent 1.39

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.79

oleg-nenashev committed:
Merge pull request #3024 from jglick/JTH

jenkins-test-harness 2.24 → 2.26

batmat committed:
Add missing @link

batmat committed:
Merge pull request #2892 from batmat/deprecate-timeunit2

Deprecate TimeUnit2

olivergondza committed:
Launch SSHD service asynchronously

oleg-nenashev committed:
Merge pull request #3011 from jglick/CLI-ping-JENKINS-46659

[JENKINS-46659] Avoid Jetty timeouts by sending pings

jglick committed:
[JENKINS-18114] Fixing CLI crumbs (#3019)

* [JENKINS-18114] The CLI client already asks for a crumb; we just to actually serve it, from CrumbIssuer.RestrictedApi.

* serveCliActionToAnonymousUserWithoutPermissions() was checking some stuff better checked by authentication(), and was failing since this fake client was not passing a crumb.

* Bring back CliCrumbExclusion, needed for anonymous use with no read access, and just let the client stop asking for a crumb.

* Added FullDuplexHttpServiceTest; useful to have a simple, self-contained demonstration of the HTTP Duplex transport.

Commit:
[Fix JENKINS-42577] Save jenkins version in Jenkins.save() (#3010)

* [Fix JENKINS-42577] Save jenkins version in Jenkins.save()

* [Fix JENKINS-42577] Fix unit test failures

Commit:
[Fix JENKINS-23411] Throws Failure if the item does not exist (#3007)

* [Fix JENKINS-23411] Throws Failure if the item does not exist

* [Fix JENKINS-23411] Adress wording in error message

oleg-nenashev committed:
Merge pull request #3027 from jglick/HudsonPrivateSecurityRealmTest

Consolidate HudsonPrivateSecurityRealmTest & HudsonPrivateSecurityRealm2Test

oleg-nenashev committed:
Merge pull request #3020 from jglick/ysoserial

Suppress compiler warnings about non-API classes used by gadgets

LinuxSuRen committed:
remove unused imports (#3032)

* del unused imports

* del unused imports

thomasgl-orange committed:
make POST requests when checking AbstractCommandInstaller.command (textarea)

rsandell committed:
[JENKINS-44657] Move sub-job's buildsdir when outside of itemrootdir (#2932)

* A test to verify JENKINS-44657 is working in core

* [JENKINS-44657] Move sub-job's buildsdir when outside of itemrootdir.

* [JENKINS-44657] Fix UsageStatisticsTest

Because we've added a test dependency that affects it.

* [JENKINS-44657] Normalize the path before comparing

* [JENKINS-44657] Use MockFolder instead of folders-plugin

for tests

* [JENKINS-44657] Document and annotate isDefaultBuildDir

* [JENKINS-44657] isDescendant doesn't throw IOException any more

* [JENKINS-44657] catch InvalidPathException and rethrow as IOException

liketic committed:
[Fix JENKINS-23411] Fix testing failure introduced in PR-3007

oleg-nenashev committed:
Merge pull request #3039 from liketic/bugfix/pull/3007

[Fix JENKINS-23411] Fix testing failure introduced in PR-3007

olivergondza committed:
Merge branch 'master' into JENKINS-19473

olivergondza committed:
Merge pull request #2976 from olivergondza/JENKINS-19473

[FIXED JENKINS-21905] Present human readable error when archiving unreadable artifacts

olivergondza committed:
Merge branch 'master' of github.com:jenkinsci/jenkins

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.80

LinuxSuRen committed:
add help-name-zh.html

daniel-beck committed:
[JENKINS-32623] Use real URL for user

daniel-beck committed:
[JENKINS-32623] Fix link to user profile in title

daniel-beck committed:
[JENKINS-40344] Don't log warning for invalid crumb from anon

daniel-beck committed:
[JENKINS-40344] Don't log second warning for anon either

jglick committed:
[FIXED JENKINS-47139] Do not set Jenkins.version too early in startup, lest InstallUtil.getLastExecVersion get confused.

batmat committed:
[JENKINS-47127] Downgrade to m-war-p 3.0.0

MWAR-404 introduced the regression in maven-war-plugin 3.1.0that makes
every files be filtered, including binaries.

MWAR-407 has been filed to revert MWAR-404, but that will be only in
m-war-p 3.1.1 or 3.2.0 which are not yet released.

So we downgrade to 3.0.0 for now.

jglick committed:
Merge pull request #3050 from jglick/setup-wizard-version-JENKINS-47139

[JENKINS-47139] Fix setup wizard

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.81

daniel-beck committed:
[JENKINS-40344] Log the user whose crumb was invalid

daniel-beck committed:
[JENKINS-40344] Fix check for anonymous authentication

Same check as User.get(Authentication) uses, so this should work

daniel-beck committed:
[JENKINS-45472] Add test

Without the fix, test fails in line 308 with:

org.junit.ComparisonFailure:
Expected :for p2
Actual   :for p1

jglick committed:
[JENKINS-44657] Noting when #2932 was merged.

jglick committed:
jenkins-test-harness 2.28

daniel-beck committed:
Merge pull request #3054 from LinuxSuRen/suren-zh

Add Chinese translation

daniel-beck committed:
Merge pull request #3052 from batmat/JENKINS-47127-revert-to-m-war-p-3.0

[JENKINS-47127] Downgrade to m-war-p 3.0.0

jglick committed:
Merge pull request #3053 from jglick/JTH

jenkins-test-harness 2.28

oleg-nenashev committed:
Merge pull request #3028 from daniel-beck/JENKINS-45472

[JENKINS-45472] Copy the list in case it's reused elsewhere

batmat committed:
[JENKINS-44052] Document & fix intended behaviour (#3043)

* [JENKINS-44052] Document & fix intended behaviour

The unit was actually ignored...

* Add tests for getTimeInSeconds

batmat committed:
Add @Issue to reference JENKINS-44052

batmat committed:
Fix wrong Javadoc

Reported by Jesse during review, but I forgot to fix it before
the merge.

batmat committed:
Add missing @since TODO

batmat committed:
Merge pull request #3058 from batmat/JENKINS-44052-followup

[JENKINS-44052] Small followups on #3043

daniel-beck committed:
Merge pull request #3049 from daniel-beck/JENKINS-40344

[JENKINS-40344] Don't log warning for invalid crumb from anon

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.82

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

olivergondza committed:
Do not clutter logs with computer update times

daniel-beck committed:
Merge pull request #3046 from daniel-beck/JENKINS-32623

[JENKINS-32623] Use real URL for user

daniel-beck committed:
Merge pull request #3037 from thomasgl-orange/AbstractCommandInstaller-textarea-checkMethod-POST

make POST requests when checking AbstractCommandInstaller.command

olivergondza committed:
Merge pull request #3040 from olivergondza/kill-update-node-time-tracking

Do not clutter logs with computer update times

oleg-nenashev committed:
Merge pull request #3031 from olivergondza/async-sshd

Launch SSHD service asynchronously
